cpp/src/parquet/encoding.cc

```diff
@@ -1500,9 +1506,18 @@ class PlainByteArrayDecoder : public PlainDecoder<ByteArrayType>,
     RETURN_NOT_OK(helper.builder->ReserveData(
         std::min<int64_t>(len_, helper.chunk_space_remaining)));
     for (int i = 0; i < num_values; ++i) {
-      int32_t value_len = static_cast<int32_t>(arrow::util::SafeLoadAs<uint32_t>(data_));
-      int increment = static_cast<int>(sizeof(uint32_t) + value_len);
-      if (ARROW_PREDICT_FALSE(len_ < increment)) ParquetException::EofException();
+      // For compiler warnings on unsigned/signed arithmetic.
```

Comment:
`ReadByteArray` helper proposed up there can hopefully simplify the next 4 functions.